Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/GenericConstraintSolver.cpp

```diff
@@ -1119,6 +1151,222 @@ void GenericConstraintProblem::unbuiltGaussSeidel(SReal timeout, GenericConstrai
     }
 }
 
+void GenericConstraintProblem::NNCG(GenericConstraintSolver* solver, int iterationNewton)
+{
+    if(!solver)
+        return;
+    int dimension = getDimension();
+    int maxIterations = iterationNewton;
+
+    if(!dimension)
+    {
+        currentError = 0.0;
+        currentIterations = 0;
+        return;
+    }
+
+
+    lam.clear();
+    lam.resize(dimension);
+    deltaF.clear();
+    deltaF.resize(dimension);
+    deltaF_new.clear();
+    deltaF_new.resize(dimension);
+    p.clear();
+    p.resize(dimension);
+
+
+    double *dfree = getDfree();
+    double *force = getF();
+    double **w = getW();
+    double tol = tolerance;
+
+    double *d = _d.ptr();
+
+    int i, j, k, l, nb;
```

Comment:
I prefer to declare the loop variables directly inside the `for` directive. I think it's less error-prone